system-addon/lib/TelemetrySender.jsm

```diff
@@ -0,0 +1,126 @@
+/* globals Preferences, Services, XPCOMUtils */
+
+const {interfaces: Ci, utils: Cu} = Components;
+
+Cu.import("resource://gre/modules/Preferences.jsm");
+Cu.importGlobalProperties(["fetch"]);
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+const ENDPOINT_PREF = "telemetry.ping.endpoint";
+const TELEMETRY_PREF = "telemetry";
+const LOGGING_PREF = "performance.log";
+
+const ACTION_NOTIF = "user-action-event";
+const PERFORMANCE_NOTIF = "performance-event";
+const COMPLETE_NOTIF = "tab-session-complete";
```

Comment:
We should sync up with @k88hudson to see how she's planning to handle session start and session complete pings - it's possible this notification will look different, or we may not need it if we use the RemotePageManager unload events, or we may have to add a ```tab-sessions-start``` event since we're no longer using tab open as an indicator